src/articulated_vehicles.cpp

```diff
@@ -143,9 +140,10 @@ CargoArray GetCapacityOfArticulatedParts(EngineID engine)
 	CargoArray capacity{};
 	const Engine *e = Engine::Get(engine);
 
-	CargoID cargo_type;
-	uint16_t cargo_capacity = GetVehicleDefaultCapacity(engine, &cargo_type);
-	if (cargo_type < NUM_CARGO) capacity[cargo_type] = cargo_capacity;
+	{
+		auto [cargo_type, cargo_capacity] = GetVehicleDefaultCapacity(engine);
+		if (IsValidCargoID(cargo_type)) capacity[cargo_type] = cargo_capacity;
+	}
```

Comment:
In other places we don't change scope, but we use `std::tie()` on next calls.

Reply:
Is that better? I think `std::tie()` would make sense if we specifically wanted to place the values into the existing variables. Here we don't care, they just happen to be the same name. And scoping means we can't accidentally use the values from earlier.

Reply:
What about something like this? It's scoped inside the `if()` instead:

```c++
	if (auto [cargo, cap] = GetVehicleDefaultCapacity(engine); IsValidCargoID(cargo) && cap > 0) {
		SetBit(cargoes, cargo);
	}
```